### PR TITLE
Spring Boot CRUD veterinary. Data JPA, Exceptions and Testing Mockito

### PR DESCRIPTION
**Pull request by mike-angek96**
Veterinary.
Customer and Animal
- Spring Data JPA
- Exception Handler
- Test with Mockito and Junit
- Rest API 
- CRUD

### CODE REVIEW

File: MiguelAngelVidalLiborio/Dia9-02Septiembre/vet/src/main/java/com/veterinary/exception/ModelNotFoundException.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package com.veterinary.exception;
+
+public class ModelNotFoundException extends RuntimeException {
```
Comment (luisod78) [inline MiguelAngelVidalLiborio/Dia9-02Septiembre/vet/src/main/java/com/veterinary/exception/ModelNotFoundException.java:3]: It's not so common to extend from RuntimeExc... but I think this is a very good example... if there is no DB, how to recover from that kind of err.... good!

PS. Maybe when app is loading context, you should validate that DB already exists... only a comment, not a requirement for this excercise

File: MiguelAngelVidalLiborio/Dia9-02Septiembre/vet/src/test/java/com/veterinary/VetApplicationTests.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.context.SpringBootTest;
+
+@SpringBootTest
+class VetApplicationTests {
+
+	@Test
+	void contextLoads() {
+	}
```
Comment (luisod78) [inline MiguelAngelVidalLiborio/Dia9-02Septiembre/vet/src/test/java/com/veterinary/VetApplicationTests.java:11]: An empty test???
A test without an assert is not a test at all....

File: MiguelAngelVidalLiborio/Dia9-02Septiembre/vet/src/test/java/com/veterinary/controller/AnimalControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+
+import com.veterinary.dto.AnimalDTO;
+import com.veterinary.dto.ResponseDTO;
+import com.veterinary.entity.Animal;
+import com.veterinary.serviceimp.AnimalServiceImp;
+
+@RunWith(MockitoJUnitRunner.class)
+public class AnimalControllerTest {
```
Comment (luisod78) [inline MiguelAngelVidalLiborio/Dia9-02Septiembre/vet/src/test/java/com/veterinary/controller/AnimalControllerTest.java:25]: nice!

File: MiguelAngelVidalLiborio/Dia9-02Septiembre/vet/src/test/java/com/veterinary/mapper/AnimalMapperTest.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+		animalDTO.setName("pikachu");
+		animalDTO.setType("rat");
+		animalDTO.setAge(4);
+
+		when(animalMapper.entityToAnimalDTO(animalEntity)).thenReturn(animalDTO);
+
+		AnimalDTO response = animalMapper.entityToAnimalDTO(animalEntity);
+
+		assertNotNull(response);
+
```
Comment (luisod78) [inline MiguelAngelVidalLiborio/Dia9-02Septiembre/vet/src/test/java/com/veterinary/mapper/AnimalMapperTest.java:42]: not even a single one assertEquals?

File: MiguelAngelVidalLiborio/Dia9-02Septiembre/vet/src/test/java/com/veterinary/mapper/CustomerMapperTest.java [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+		customerDTO.setId(1L);
+		customerDTO.setFullName("Leo Messi");
+		customerDTO.setAddress("Paris");
+		customerDTO.setPayment(800);
+
+		Mockito.when(customerMapper.entityToCustomerDTO(customerEntity)).thenReturn(customerDTO);
+
+		CustomerDTO response = customerMapper.entityToCustomerDTO(customerEntity);
+
+		assertNotNull(response);
```
Comment (luisod78) [inline MiguelAngelVidalLiborio/Dia9-02Septiembre/vet/src/test/java/com/veterinary/mapper/CustomerMapperTest.java:39]: assertEquals?? none?

**Review by luisod78 — CHANGES_REQUESTED** [review]

File: MiguelAngelVidalLiborio/Dia9-02Septiembre/vet/src/main/java/com/veterinary/exception/EmptyInputException.java [hunk 6]
```diff
@@ -0,0 +1,10 @@
+package com.veterinary.exception;
+
+public class EmptyInputException extends RuntimeException {
```
[the remark this answers is not in the document]
Comment (mike-angek96) [reply MiguelAngelVidalLiborio/Dia9-02Septiembre/vet/src/main/java/com/veterinary/exception/EmptyInputException.java:3]: interesting i did not know that, the tutorials from youtube always use RunTimeException, thanks

**Review by lrod-glb — APPROVED** [review]

**Review by globant-bootcamp — APPROVED** [review]

**Pull request merged**